relay-general/src/protocol/replay.rs

```diff
@@ -256,6 +256,27 @@ impl Replay {
         self.normalize_ip_address(client_ip);
         self.normalize_user_agent(user_agent);
         self.normalize_type();
+        self.normalize_array_fields();
+    }
+
+    fn normalize_array_fields(&mut self) {
+        let new_error_ids = self.error_ids.value_mut().as_mut().map(|items| {
+            items.truncate(100);
+            items.to_owned()
+        });
+        self.error_ids.set_value(new_error_ids);
+
+        let new_trace_ids = self.trace_ids.value_mut().as_mut().map(|items| {
+            items.truncate(100);
+            items.to_owned()
+        });
+        self.trace_ids.set_value(new_trace_ids);
+
+        let new_urls = self.urls.value_mut().as_mut().map(|items| {
+            items.truncate(100);
+            items.to_owned()
+        });
+        self.urls.set_value(new_urls);
```

Comment:
So if you would like to truncate the items, you could do it directly without copying the data, since you have here a `&mut` to that list.  
I think something like following should work:

```suggestion
        if let Some(items) = self.error_ids.value_mut() {
            items.truncate(100);
        }

        if let Some(items) = self.trace_ids.value_mut() {
            items.truncate(100);
        }

        if let Some(items) = self.urls.value_mut() {
            items.truncate(100);
        }

```